edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/NestedDeviceScopeApiClient.cs

```diff
@@ -126,6 +151,38 @@ async Task<ScopeResult> GetIdentitiesInTargetScopeWithRetry(Uri uri, Option<stri
             }
         }
 
+        async Task<ScopeResult> GetIdentityOnBehalfOf(Uri uri, string deviceId, Option<string> moduleId)
+        {
+            HttpClient client = this.proxy
+                .Map(p => new HttpClient(new HttpClientHandler { Proxy = p }, disposeHandler: true))
+                .GetOrElse(() => new HttpClient());
+            using (var msg = new HttpRequestMessage(HttpMethod.Post, uri))
+            {
+                // Get the auth-chain for the target device
+                Option<string> maybeAuthChain = await this.serviceIdentityHierarchy.GetEdgeAuthChain(deviceId);
+                string authChain = maybeAuthChain.Expect(() => new InvalidOperationException($"No valid authentication chain for {deviceId}"));
+
+                var payload = new IdentityOnBehalfOfRequest(deviceId, moduleId.OrDefault(), authChain);
+                string token = await this.edgeHubTokenProvider.GetTokenAsync(Option.None<TimeSpan>());
```

Comment:
Does this means it awaits forever? If we lose connection does that mean threads will accumulate here?

Reply:
this is not the timeout value, it's the expiration for the token.  Passing in None results in the default expiration being used.

---
In reply to: [457529858](https://github.com/Azure/iotedge/pull/3248#discussion_r457529858) [](ancestors = 457529858)